hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp

```diff
@@ -316,13 +316,40 @@ namespace hpx { namespace threads { namespace detail
     }
 
     template <typename Scheduler>
-    void scheduled_thread_pool<Scheduler>::resume(error_code& ec)
+    void scheduled_thread_pool<Scheduler>::suspend(error_code& ec)
     {
-        if (!(mode_ & threads::policies::enable_elasticity))
+        if (threads::get_self_ptr() && hpx::this_thread::get_pool() == this)
         {
+            HPX_THROWS_IF(ec, bad_parameter,
+                "scheduled_thread_pool<Scheduler>::suspend",
+                "cannot suspend a pool from itself");
             return;
         }
 
+        if (threads::get_self_ptr())
+        {
+            while (sched_->Scheduler::get_thread_count() >
+                get_background_thread_count())
+            {
+                hpx::this_thread::suspend();
+            }
+        }
+        else
+        {
+            while (sched_->Scheduler::get_thread_count() >
+                get_background_thread_count())
+            {}
```

Comment:
Should this at least call `sleep(0)` instead of unconditionally spinning? Alternatively, we could re-spawn the resume function on an HPX thread instead of spinning, not sure...

Reply:
`sleep(0)` (or `yield`) is a good idea. HPX thread I'm not sure about. It would require returning a future (if we want to know when it's done), and so far I've assumed this can be called from outside the runtime.

Reply:
See here for how this could be handled: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/util/detail/yield_k.hpp. In fact, you could use `yield_k()` instead of having your own loop.

Reply:
Updated to use `yield_k` - made most waits nice and short. Thanks for pointing that out.